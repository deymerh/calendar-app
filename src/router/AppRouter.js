import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  BrowserRouter as Router,
  Switch,
  Redirect
} from 'react-router-dom';
import { startChecking } from '../actions/auth';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { PrivateRoute } from './PrivateRoutes';
import { PublicRoute } from './PublicRoutes';

export const AppRouter = () => {
  const dispatch = useDispatch();
  const { checking, uid } = useSelector(state => state.auth);
  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);
  if (checking) {
    return (<h4>Espere...</h4>);
  }
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            isAuthenticated={!!uid}
            path="/login"
            component={LoginScreen}
          />
          <PrivateRoute
            isAuthenticated={!!uid}
            exact
            path="/"
            component={CalendarScreen}
          />
          <Redirect to="/" />
        </Switch>
      </div>
    </Router>
  )
}
