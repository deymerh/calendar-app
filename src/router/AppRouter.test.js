import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import moment from 'moment';
import { AppRouter } from './AppRouter';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Tes in the component <AppRouter />', () => {

  test('should do Snapshot the component [<AppRouter />] with the test `Espere...` ', async () => {
    const initialState = {
      auth: {
        checking: true
      }
    }
    let store = mockStore(initialState);
    store.dispatch = jest.fn();
    const { container, getByText } = render(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );
    const text = await getByText(/Espere/);
    expect(container).toMatchSnapshot();
    expect(text).toBeInTheDocument();
  });

  test('should to show the public route', async () => {
    const initialState = {
      events: [
        {
          id: new Date().getTime(),
          title: 'Mi cumpleaños',
          start: moment().toDate(),
          end: moment().add(2, 'hours').toDate(),
          bgColor: '#fafafa',
          user: {
            _id: '12345',
            name: 'Deymer'
          }
        }
      ],
      eventActive: null,
      auth: {
        checking: false,
        uid: null
      }
    }
    let store = mockStore(initialState);
    store.dispatch = jest.fn();
    const { container, getByText } = render(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );
    const text = await getByText(/Ingreso/);
    expect(text).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('should to show the public route', async () => {
    const initialState = {
      ui: {
        modalOpen: false
      },
      calendar: {
        events: [
          {
            id: new Date().getTime(),
            title: 'Mi cumpleaños',
            start: moment().toDate(),
            end: moment().add(2, 'hours').toDate(),
            bgColor: '#fafafa',
            user: {
              _id: '12345',
              name: 'Deymer'
            }
          }
        ],
      },
      eventActive: null,
      auth: {
        checking: false,
        uid: '12345',
        name: 'DeymerH'
      }
    }
    let store = mockStore(initialState);
    store.dispatch = jest.fn();
    const { container, getByText } = render(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );
    const text = await getByText(/DeymerH/);
    expect(text).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

});