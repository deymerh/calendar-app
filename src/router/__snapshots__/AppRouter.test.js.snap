// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tes in the component <AppRouter /> should do Snapshot the component [<AppRouter />] with the test \`Espere...\`  1`] = `
<div>
  <h4>
    Espere...
  </h4>
</div>
`;

exports[`Tes in the component <AppRouter /> should to show the public route 1`] = `
<div>
  <div>
    <div
      class="container login-container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6 login-form-1"
        >
          <h3>
            Ingreso
          </h3>
          <form>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                name="lEmail"
                placeholder="Correo"
                type="text"
                value="deymerh@hotmail.com"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                name="lPassword"
                placeholder="Contraseña"
                type="password"
                value="123456"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="btnSubmit"
                type="submit"
                value="Login"
              />
            </div>
          </form>
        </div>
        <div
          class="col-md-6 login-form-2"
        >
          <h3>
            Registro
          </h3>
          <form>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                name="rName"
                placeholder="Nombre"
                type="text"
                value="Antonio"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                name="rEmail"
                placeholder="Correo"
                type="email"
                value="deyer1227@gmail.com"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                name="rPassword1"
                placeholder="Contraseña"
                type="password"
                value="123456"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                name="rPassword2"
                placeholder="Repita la contraseña"
                type="password"
                value="123456"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="btnSubmit"
                type="submit"
                value="Crear cuenta"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tes in the component <AppRouter /> should to show the public route 2`] = `
<div>
  <div>
    <div
      class="calendar-screen"
    >
      <div
        class="navbar navbar-dark bg-dark mb-4"
      >
        <span
          class="navbar-brand"
        >
          DeymerH
        </span>
        <button
          class="btn btn-outline-danger"
        >
          Salir
        </button>
      </div>
      <div
        class="rbc-calendar"
      >
        <div
          class="rbc-toolbar"
        >
          <span
            class="rbc-btn-group"
          >
            <button
              type="button"
            >
              Hoy
            </button>
            <button
              type="button"
            >
              &lt;
            </button>
            <button
              type="button"
            >
              &gt;
            </button>
          </span>
          <span
            class="rbc-toolbar-label"
          >
            septiembre 2021
          </span>
          <span
            class="rbc-btn-group"
          >
            <button
              class="rbc-active"
              type="button"
            >
              Mes
            </button>
            <button
              class=""
              type="button"
            >
              Semana
            </button>
            <button
              class=""
              type="button"
            >
              Día
            </button>
            <button
              class=""
              type="button"
            >
              Agenda
            </button>
          </span>
        </div>
        <div
          aria-label="Month View"
          class="rbc-month-view"
          role="table"
        >
          <div
            class="rbc-row rbc-month-header"
            role="row"
          >
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                lun.
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                mar.
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                mié.
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                jue.
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                vie.
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                sáb.
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                dom.
              </span>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    30
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    31
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    01
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    02
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    03
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    04
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    05
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg rbc-today"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    06
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    07
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-now rbc-current"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    08
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    09
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    10
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    11
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    12
                  </a>
                </div>
              </div>
              <div
                class="rbc-row"
              >
                <div
                  class="rbc-row-segment"
                  style="flex-basis: 28.57142857142857%; max-width: 28.57142857142857%;"
                >
                   
                </div>
                <div
                  class="rbc-row-segment"
                  style="flex-basis: 14.285714285714285%; max-width: 14.285714285714285%;"
                >
                  <div
                    class="rbc-event"
                    style="background-color: rgb(70, 86, 96); border-radius: 0px; opacity: 0.8; display: block; color: white;"
                    tabindex="0"
                  >
                    <div
                      class="rbc-event-content"
                      title="Mi cumpleaños"
                    >
                      <div>
                        <span>
                          Mi cumpleaños
                        </span>
                        <strong>
                           - 
                          Deymer
                        </strong>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    13
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    14
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    15
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    16
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    17
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    18
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    19
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    20
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    21
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    22
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    23
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    24
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    25
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    26
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    27
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    28
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    29
                  </a>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    30
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    01
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    02
                  </a>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <a
                    href="#"
                    role="cell"
                  >
                    03
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="btn btn-primary fab"
      >
        <i
          class="fas fa-plus"
        />
      </button>
    </div>
  </div>
</div>
`;
